posthog-web/src/context.ts

```diff
@@ -1,9 +1,9 @@
 import { utils } from '../../posthog-core'
 import { version } from '../package.json'
 
-export function getContext(window: Window): any {
+export function getContext(window: Window | undefined): any {
   let context = {}
-  if (window.navigator) {
+  if (window && window.navigator) {
```

Comment:
Most of these can be better written as `window?.navigator` 
```suggestion
  if (window?.navigator) {
```